python/ql/lib/semmle/python/dataflow/new/internal/ImportResolution.qll

```diff
@@ -64,6 +64,19 @@ private import semmle.python.dataflow.new.internal.DataFlowPrivate
  *   `bar` subpackage.
  */
 module ImportResolution {
+  /**
+   * Holds if there is an ESSA step from `defFrom` to `defTo`, which should be allowed
+   * for import resolution.
+   */
+  private predicate allowedEssaImportStep(EssaDefinition defFrom, EssaDefinition defTo) {
+    // to handle definitions guarded by if-then-else
+    defFrom = defTo.(PhiFunction).getAnInput()
+    or
+    // refined variable
+    // example: https://github.com/nvbn/thefuck/blob/ceeaeab94b5df5a4fe9d94d61e4f6b0bbea96378/thefuck/utils.py#L25-L45
```

Comment:
That's a... Colourful project name.

Reply:
I wonder if `Foo` is kinda-colourful as well. foo/bar = [FUBAR](https://dictionary.cambridge.org/dictionary/english/fubar)